File: server/src/main/java/com/topo/model/entity/Topology.java
package com.topo.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 拓扑实体 — 对应 tb_topology 表
 *
 * 画布拓扑（设备+连线）的 MySQL 持久化存储：
 * topologyJson 直接存 TopologyJson 序列化字符串（MySQL JSON 列），
 * 每个用户可保存多个拓扑，删除走逻辑删除。
 */
@Data
@TableName("tb_topology")
public class Topology {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;         // 所属用户

    private String name;         // 拓扑名称

    private String topologyJson; // 画布结构 JSON（设备/连线/接口）

    private String sourceType;   // 来源: ensp_topo_file / screenshot / manual

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;     // 0=正常, 1=已删除
}

File: client/src/api/topology.ts
import request from './request'

/** 拓扑持久化接口（MySQL tb_topology） */

export interface TopologySaveData {
  id?: number | null
  name: string
  topologyJson: string
  sourceType?: string
}

/** 我的拓扑列表（按更新时间倒序，不含完整 JSON） */
export function listTopology() {
  return request.get('/topology/list')
}

/** 拓扑详情（含完整 topologyJson，恢复画布用） */
export function getTopology(id: number) {
  return request.get(`/topology/${id}`)
}

/** 保存拓扑：id 为空新建，否则覆盖保存 */
export function saveTopology(data: TopologySaveData) {
  return request.post('/topology/save', data)
}

/** 删除拓扑（逻辑删除，仅限本人） */
export function deleteTopology(id: number) {
  return request.delete(`/topology/${id}`)
}
